streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -115,8 +162,189 @@ public void shutdown() throws IOException {
         IntegrationTestUtils.purgeLocalStreamsState(streamsConfiguration);
     }
 
+
+    /**
+     * Creates a typical word count topology
+     * @param inputTopic
+     * @param outputTopic
+     * @param streamsConfiguration config
+     * @return
+     */
+    private KafkaStreams createCountStream(String inputTopic, String outputTopic, Properties streamsConfiguration) {
+        KStreamBuilder builder = new KStreamBuilder();
+        final Serde<String> stringSerde = Serdes.String();
+        final KStream<String, String> textLines = builder.stream(stringSerde, stringSerde, inputTopic);
+
+        final KGroupedStream<String, String> groupedByWord = textLines
+            .flatMapValues(new ValueMapper<String, Iterable<String>>() {
+                @Override
+                public Iterable<String> apply(String value) {
+                    return Arrays.asList(value.toLowerCase(Locale.getDefault()).split("\\W+"));
+                }
+            })
+            .groupBy(MockKeyValueMapper.<String, String>SelectValueMapper());
+
+        // Create a State Store for the all time word count
+        groupedByWord.count("word-count-store-" + inputTopic).to(Serdes.String(), Serdes.Long(), outputTopic);
+
+        // Create a Windowed State Store that contains the word count for every 1 minute
+        groupedByWord.count(TimeWindows.of(60000), "windowed-word-count-store-" + inputTopic);
+
+        return new KafkaStreams(builder, streamsConfiguration);
+    }
+
+    private class StreamRunnable implements Runnable {
+        private final KafkaStreams myStream;
+        private final String inputTopic;
+        private final String outputTopic;
+        private final int queryPort;
+
+        StreamRunnable(String inputTopic, String outputTopic, int queryPort) {
+            this.inputTopic = inputTopic;
+            this.outputTopic = outputTopic;
+            this.queryPort = queryPort;
+            Properties props = (Properties) streamsConfiguration.clone();
+            props.put(StreamsConfig.APPLICATION_SERVER_CONFIG, "localhost:" + queryPort);
+            this.myStream = createCountStream(inputTopic, outputTopic, props);
+        }
+
+        @Override
+        public void run() {
+            this.myStream.start();
+
+        }
+
+        public void close() {
+            this.myStream.close();
+        }
+
+        public final KafkaStreams getStream() {
+            return myStream;
+        }
+    }
+
+    private void verifyAllKVKeys(StreamRunnable[] streamRunnables, KafkaStreams streams,
```

Comment:
the comment `// store might have closed but rebalance hasn't happened` should have said:
`// kafka streams instance may have closed but rebalance hasn't happened`